explorer/data/prelude.carbon

```diff
@@ -416,6 +416,56 @@ external impl i32 as RightShiftWith(i32) where .Result == i32 {
   }
 }
 
+//-------------------------
+// Optional
+//-------------------------
+choice OptionalElement(T:! Type) {
+  None(),
+  Element(T)
+}
+
+class Optional(T:! Type){
+  fn CreateEmpty() -> Optional(T){
+    return { .element = OptionalElement(T).None() };
+  }
+  fn Create ( value: T ) -> Optional(T){
+    return { .element = OptionalElement(T).Element(value) };
+  }
+
+  fn has_value[me: Self] () -> bool{
+    var x: OptionalElement(T) = me.element;
+    match(x){
+      case OptionalElement(T).None() => { return false; }
+    }
+    return true;
+  }
+
+  fn get[me: Self] () -> T {
+    var y: T;
+    var x: OptionalElement(T) = me.element;
+    match(x){
+      case OptionalElement(T).Element( var x: T ) =>{
+        return x;
+      }
+    }
+    return y;
```

Comment:
It'd be nice to assert or something on this codepath. Maybe we don't have a way of doing that yet, though?

Reply:
To make it reasonable, we need some form of error handling, which does not yet exist.
For this reason, only this possibility remains.

Reply:
I don't think we need error handling: assertion failure can just terminate the program. It shouldn't be too hard to add an intrinsic for that.

At the very least, there should be a TODO here to add an assertion.

Reply:
TODO is added and a new PR for intrinsic_assert is in work